langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal

```diff
@@ -1907,6 +1924,23 @@ function testArrayPop() {
         assertValueEquality("pop() not supported on type 'int[2]'",
         <string> checkpanic result.detail()["message"]);
     }
+
+    [int, string, int...] tuple1 = [1, "hello", 4];
+    int|string result2 = tuple1.pop();
+    assertTrue(result2 is int);
+    assertValueEquality(4, result2);
+    int|string|error result3 = trap tuple1.pop();
+    assertTrue(result3 is error);
+    if (result3 is error) {
+        assertValueEquality("{ballerina/lang.array}OperationNotSupported", result3.message());
+        assertValueEquality("the number of members in a tuple value should be greater than the member types of the tuple to perform this operation",
+        <string> checkpanic result3.detail()["message"]);
+    }
+
+    [int, string, Person2...] tuple2 = [1, "hello", {id:5, name:"John", age:56}];
```

Comment:
Shall we add a test case for `pop()` in empty tuples, Recursive tuples, tuple with type reference type

Reply:
I'll add those.